GpsDataAnalyzer/visualizer/visualizer.py

```diff
@@ -0,0 +1,139 @@
+            elif abs_deviation > 5 and abs_deviation <= 10:
+                confidence.append(2)
+            else:
+                confidence.append(1)
+
+        deviation_dataframe["Confidence"] = confidence 
+
+        return deviation_dataframe
+
+
+    def draw_hist_graph(self, data, x_label, y_label, title, availability):
+        """
+        Draw the histogram graph and save it as a png file
+
+        Args:
+          data: data on y axis
+          x_label: label for x axis
+          y_label: label for y axis
+          title: title for the graph
+          availability: percentile of captured datapoints
+        """
+        # Plot the data
+        fig = plt.figure(figsize=(20,10))
+        hist_label = 'Availability: ' + str(availability) + '%'
+        plt.hist(data, align='mid', bins=[0.5,1.5,2.5,3.5], rwidth=0.8, label=hist_label, orientation="horizontal")
+        
+        # Set the title and labels    
+        plt.legend(loc="upper left")
+        plt.xlabel(x_label, fontsize=10)
+        plt.ylabel(y_label, fontsize=10)
+        plt.title(title, fontsize=12)
+        plt.yticks(range(0,5))
+
+        # Save the graph as a png picture
+        my_path = os.path.dirname(__file__)
+        my_file = title + "_" + datetime.strftime(datetime.now(), "%Y-%m-%dT%H:%M:%S") + ".png"
+        fig.savefig(os.path.join(my_path, my_file)) 
+
+
+    def draw_line_graph(self, x_data, x_label, y_data, y_label, title):
+        """
+        Draw the line graph and save it as a png file
+
+        Args:
+          x_data: data on x axis
+          x_label: label for x axis
+          y_data: data on y axis
+          y_label: label for y axis
+          title: title for the graph
+        """
+        # Get the absolute mean of deviation, stadard deviation, min and max deviation
+        abs_mean_deviation = round(np.mean(y_data),3)
+        std_deviation = round(np.std(y_data),3)
+        min_deviation = round(self.get_min_deviation(y_data), 3)
+        max_deviation = round(self.get_max_deviation(y_data), 3)
+
+        # Get the time duration 
+        time_duration = x_data[len(x_data) - 1] - x_data[0]
+
+        # Set the line_label and x_label
+        line_label = 'Mean: '+ str(abs_mean_deviation) + '\n' +'STD: ' + str(std_deviation) + '\n' + 'Min: ' + str(min_deviation) + '\n' + 'Max: ' + str(max_deviation)
+        x_label += str(time_duration)
+
+        # Plot the data
+        fig = plt.figure(figsize=(20,10))
+        ax = plt.subplot()
+        ax.plot(x_data, y_data, label= line_label)
+
+        # Format the time on x axis '%H:%M:%S'
+        ax.xaxis.set_major_formatter(dates.DateFormatter('%H:%M:%S'))
+
+        # Set the title and labels
+        plt.legend(loc="upper left")
+        plt.title(title, fontsize = 12)
+        plt.xlabel(x_label, fontsize = 10)
+        plt.ylabel(y_label, fontsize = 10)
+
+        # Save the graph as a png picture
+        my_path = os.path.dirname(__file__)
```

Comment:
Will this save the picture in the same folder as the code?

Reply:
Yes, it will be saved to the same folder.